cli/cmd/generate_aws.go

```diff
@@ -0,0 +1,349 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"time"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/lacework/go-sdk/lwgenerate/aws"
+	"github.com/pkg/errors"
+)
+
+var (
+	// Define question text here so they can be reused in testing
+	QuestionEnableConfig                = "Enable Config Integration?"
+	QuestionEnableCloudtrail            = "Enable Cloudtrail Integration?"
+	QuestionAwsRegion                   = "Specify the AWS region Cloudtrail, SNS, and S3 resources should use:"
+	QuestionConsolidatedCloudtrail      = "Use consolidated Cloudtrail?"
+	QuestionUseExistingCloudtrail       = "Would you like to use an existing Cloudtrail?"
+	QuestionCloudtrailExistingBucketArn = "Specify an existing bucket ARN used for Cloudtrail logs:"
+	QuestionForceDestroyS3Bucket        = "Should the new S3 bucket have force destroy enabled?"
+	QuestionExistingIamRoleName         = "Specify an existing IAM role name for Cloudtrail access"
+	QuestionExistingIamRoleArn          = "Specify an existing IAM role ARN for Cloudtrail access"
+	QuestionExistingIamRoleExtId        = "Specify the external ID to be used with the existing IAM role"
+	QuestionPrimaryAwsAccountProfile    = "Before adding subaccounts, your primary AWS account profile name must be set; which profile should the main account use?"
+	QuestionSubAccountProfileName       = "Supply the profile name for this additional AWS account:"
+	QuestionSubAccountRegion            = "What region should be used for this account?"
+	QuestionSubAccountAddMore           = "Add another AWS account?"
+	QuestionConfigAdvanced              = "Configure advanced integration options?"
+	QuestionAnotherAdvancedOpt          = "Configure another advanced integration option"
+	QuestionCustomizeOutputLocation     = "Provide the location for the output to be written:"
+
+	// select options
+	AdvancedOptDone        = "Done"
+	AdvancedOptCloudTrail  = "Additional Cloudtrail Options"
+	AdvancedOptIamRole     = "Configure Lacework integration with an existing IAM role"
+	AdvancedOptAwsAccounts = "Add Additional AWS Accounts to Lacework"
+	AdvancedOptLocation    = "Customize Output Location"
+)
+
+// survey.Validator for aws ARNs
+//
+// This isn't service/type specific but rather just validates that an ARN was entered that matches valid ARN formats
+func validAwsArnFormat(val interface{}) error {
+	// the reflect value of the result
+	value := reflect.ValueOf(val)
+
+	// if the value passed is not a string
+	if value.Kind() != reflect.String {
+		return errors.New("value must be a string")
+	}
+
+	// if value doesn't match regex, return invalid arn
+	// original source: https://regex101.com/r/pOfxYN/1
+	matchRegEx := `^arn:(?P<Partition>[^:\n]*):(?P<Service>[^:\n]*):(?P<Region>[^:\n]*):(?P<AccountID>[^:\n]*):(?P<Ignore>(?P<ResourceType>[^:\/\n]*)[:\/])?(?P<Resource>.*)$`
```

Comment:
Would very much like a unit test for this regex

Reply:
👍  will do

Reply:
Is this a REGEX you have generated, or was this pulled from AWS docs? If the latter, could we get a source link?

Reply:
@rmoles the source link is in the comment above this line